org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/NexusSlit.java

```diff
@@ -0,0 +1,254 @@
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with GDA. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eclipse.scanning.example.scannable;
+
+import java.text.MessageFormat;
+
+import org.eclipse.dawnsci.nexus.INexusDevice;
+import org.eclipse.dawnsci.nexus.NXobject;
+import org.eclipse.dawnsci.nexus.NXslit;
+import org.eclipse.dawnsci.nexus.NexusBaseClass;
+import org.eclipse.dawnsci.nexus.NexusException;
+import org.eclipse.dawnsci.nexus.NexusNodeFactory;
+import org.eclipse.dawnsci.nexus.NexusScanInfo;
+import org.eclipse.dawnsci.nexus.NexusScanInfo.ScanRole;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectProvider;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectWrapper;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.ILazyWriteableDataset;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.scanning.api.AbstractScannable;
+import org.eclipse.scanning.api.IScanAttributeContainer;
+import org.eclipse.scanning.api.IScannable;
+import org.eclipse.scanning.api.annotation.scan.ScanFinally;
+import org.eclipse.scanning.api.event.scan.DeviceValueMultiPosition;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+import org.eclipse.scanning.example.Services;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class provides an implementation which will write an NXslit to NeXus based on information from individual scannables
+ */
+public class NexusSlit extends AbstractScannable<DeviceValueMultiPosition> implements INexusDevice<NXslit> {
+
+	private static final Logger logger = LoggerFactory.getLogger(NexusSlit.class);
+
+	private String x_gap_deviceName;
+	private String y_gap_deviceName;
+
+	private IScannable<?> x_gap;
+	private IScannable<?> y_gap;
+
+	public NexusSlit() {
+		super(Services.getScannableDeviceService());
+	}
+
+	// implements IScannable<DeviceValueMultiPosition>
+
+	@Override
+	public DeviceValueMultiPosition getPosition() throws Exception {
+		DeviceValueMultiPosition position = new DeviceValueMultiPosition();
+		position.put(NXslit.NX_X_GAP, (double)x_gap.getPosition());
+		position.put(NXslit.NX_Y_GAP, (double)y_gap.getPosition());
+		return position;
+	}
+
+	@Override
+	public void setPosition(DeviceValueMultiPosition value, IPosition position) throws Exception {
+		logger.debug("setPosition({}, {}) called on {}", value, position, getName());
+
+		if (value!=null) {
+			logger.warn("non null setPosition() not expected on {}, ignoring...", getName());
+		}
+
+		if (position!=null) {
+			write(value, getPosition(), position);
+		}
+	}
+
+	private void write(DeviceValueMultiPosition demand, DeviceValueMultiPosition actual, IPosition loc) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)